toolkit/tools/internal/rpm/rpm.go

```diff
@@ -243,16 +247,29 @@ func executeRpmCommandRaw(program string, args ...string) (stdout string, err er
 	return
 }
 
-// DefaultDefinesWithDist returns a new map of default defines that can be used during RPM queries that also includes
-// the dist tag.
-func DefaultDefinesWithDist(runChecks bool, distTag string) map[string]string {
-	defines := DefaultDefines(runChecks)
-	defines[DistTagDefine] = distTag
+// DefaultDistroDefines returns a new map of default defines that can be used during RPM queries that also includes
+// the distro tags like '%dist', '%zal', '%alz<VER>'.
+// If distTag is empty, the distro tags will not be generated.
+func DefaultDistroDefines(runChecks bool, distTag string) map[string]string {
+	defines := defaultDefines(runChecks)
+	if distTag != "" {
+		defines[DistTagDefine] = distTag
+		// distTag is of the form ".<distro><version>", extract the distro and version
+		// This will be used to generate '%<distro> <version>'
+		distTagSplit := distTagRegex.FindStringSubmatch(distTag)
+		if len(distTagSplit) == 3 {
+			distroName := distTagSplit[1]
+			distroVersion := distTagSplit[2]
+			defines[distroName] = distroVersion
+		} else {
+			logger.Log.Warnf("Invalid distro tag (%s), won't generate derived distro defines", distTag)
+		}
```

Comment:
I'm not convinced about this part for these reasons:
- We extract `distroName` from `distTag`, so there's no guarantee it's always going to match the hard-coded value of `azl` in our specs.
- We expect the distro name and version to be set correctly for our specs to work, so the warning log is too weak in my opinion.

Ideas:
1. We could turn the warning into an error. That would be a lot of pluming changes in the code, however, and it would not help with making sure we always land with `azl` as the distro name. And I'd like to avoid hard-coding `azl`, since that would create another place, which would require an update in case of any re-branding changes in the future.
2. Since `distTag`, `distroName` and `distroVersion` are something we want to control globally for the whole repository (right now from the toolkit with a potential switch to reading the values provided by a package providing `system-release`), this looks similar to what we're doing with `exe.go::ToolkitVersion`. What do you think about compiling `distTag`, `distroName` and `distroVersion` into our tooling and completely dropping the `distTag` argument from `DefaultDistroDefines`? This would require either splitting `DIST_TAG` inside our makefiles a bit or parsing it to extract the other values like you do it here (but also immediately failing in case the format is incorrect).

Reply:
I think we still want to have DIST_TAG configurable, but the other macros I agree can be hard-coded.